sdk/profiles/src/main/java/io/opentelemetry/sdk/profiles/data/LinkData.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.profiles.data;
+
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A connection from a profile Sample to a trace Span.
+ *
+ * @see "pprofextended.proto::Link"
+ */
+@Immutable
+public interface LinkData {
+
+  /** A unique identifier of a trace that this linked span is part of. The ID is a 16-byte array. */
+  @SuppressWarnings("mutable")
+  byte[] getTraceId();
```

Comment:
🤔  you put an Immutable interface on an interface that clearly cannot be immutable. Why not use the existing TraceId/SpanId String representation from the existing API?

Reply:
Indeed, the type to be used for binary data in the API is another topic I raised in the PR text. In this specific case, it's because the span and trace fields get handled independently rather than as internal aspects of a wrapper object, which is not the same pattern as in the existing protocols. There are two related discussions there - the one about how directly the API should correspond to the underlying proto and the one about how we want to handle byte[] types in the API, since there isn't yet an established pattern for doing that. If we e.g. want to keep a very literal 1:1 between the Java API and proto fields and  we want to use ByteBuffer.asReadOnlyBuffer, then ByteBuffer may be preferred over moving the pair of fields into a wrapper structure. The existing approach is a strawman proposal - I'd be surprised if it remained unchanged.